hicommonwealth/commonwealth
# Implement admin top users

## Link to Issue
Closes: #6447

## Description of Changes
- Adds route to return list of top 150 users based on activity
- Adds UI component to admin panel to download top users list
- Refactors download logic to be reusable

## Test Plan
- Ensure the user is a super admin
- Go to http://localhost:8080/admin-panel
- At the bottom of the page, click "Generate and Download"– confirm that it downloads a CSV file
- Import CSV into Google Sheets– confirm that it renders correctly

## Deployment Plan
N/A

## Other Considerations
- While downloads are typically handled by the backend via headers,  in this implementation the client fetches JSON and converts to CSV. Not a problem since the list is small.
- JSON to CSV conversion doesn't use a library, so it doesn't catch all edgecases.
